pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -77,6 +77,7 @@ func DiffExternalMetrics(informerList []*autoscalingv2.HorizontalPodAutoscaler,
 
 // AutoscalerMetricsUpdate will return true if the applied configuration of the Autoscaler has changed.
 // We only care about updates of the metrics or their scopes.
+// We also want to process the resync events, which can be identified with the resver.
 func AutoscalerMetricsUpdate(new, old *autoscalingv2.HorizontalPodAutoscaler) bool {
-	return old.Annotations["kubectl.kubernetes.io/last-applied-configuration"] != new.Annotations["kubectl.kubernetes.io/last-applied-configuration"]
+	return old.ResourceVersion == new.ResourceVersion || old.Annotations["kubectl.kubernetes.io/last-applied-configuration"] != new.Annotations["kubectl.kubernetes.io/last-applied-configuration"]
```

Comment:
I think `new` and `old` object comes from the shared listener cache. 
so by doing: `old.Annotation["kubectl.kubernetes.io/last-applied-configuration"]` it modify the object if the key didnt exist in the map.

```
var oldAnnotationValue string
if val, ok := old.Annotation["kubectl.kubernetes.io/last-applied-configuration"]; ok { oldAnnotationValue = val}
```
same for the `new`